alghaibb/mahmoud-jaderi-portfolio
# Add custom messages component and integrate with account creation, verification, and login forms

## Description
This PR introduces a new `CustomMessage` component for displaying various types of messages (success, error, info, custom) across different forms in the application. It integrates this component into the `CreateAccountForm`, `VerifyAccountForm`, and `LoginForm` components.

## Changes
1.  **CustomMessage Component**:
    -   Added a new `CustomMessage` component with support for success, error, info, and custom messages.
    -   Includes icons and styles for each message type.
    -   Provides an `onClose` handler to allow users to dismiss messages.
2.  **CreateAccountForm**:
    -   Integrated the `CustomMessage` component for displaying success and error messages.
    -   Implemented a mechanism to clear previous messages before setting new ones.
    -   Added `onClose` handlers to the messages.
3.  **VerifyAccountForm**:
    -   Integrated the `CustomMessage` component for displaying success, error, and info messages.
    -   Implemented a mechanism to clear previous messages before setting new ones.
    -   Added `onClose` handlers to the messages.
4.  **LoginForm**:
    -   Integrated the `CustomMessage` component for displaying error messages.
    -   Implemented a mechanism to clear previous messages before setting new ones.
    -   Added `onClose` handlers to the messages.
5. **Resend Verification Server Action:**
    - Added a check in the `resend-verification-email.ts` to check if the user is verified to trigger the `info` custom message.

## How to Test
1.  **CreateAccountForm**:
    -   Navigate to the account creation page.
    -   Fill out the form with valid and invalid data to trigger success and error messages.
    -   Ensure that only one message is displayed at a time and can be dismissed.
2.  **VerifyAccountForm**:
    -   Navigate to the account verification page.
    -   Fill out the form with valid and invalid data to trigger success, error, and info messages.
    -   Ensure that only one message is displayed at a time and can be dismissed.
    -   After verifying your account, try resending a verification email again to trigger the `User is already verified` info custom message.
3.  **LoginForm**:
    -   Navigate to the login page.
    -   Fill out the form with valid and invalid data to trigger error messages.
    -   Ensure that only one message is displayed at a time and can be dismissed.
